src/Core/Framework/Api/OAuth/JWTConfigurationFactory.php

```diff
@@ -41,7 +41,7 @@ public static function createJWTConfiguration(
 
             $clock = new SystemClock(new \DateTimeZone(\date_default_timezone_get()));
 
-            $configuration->setValidationConstraints(
+            $configuration->withValidationConstraints(
```

Comment:
you need to assign it back. it creates always a new object like every ->with

Reply:
that method is new, you will need to bump up the version constraint to disallow older versions